From: Yifeng <wuyifeng@nd.com.cn>
Date: Fri, 30 Nov 2021 07:23:00 +0000
Subject: The web page forces to fill the screen when the checkbox of Force enable zoom is checked.

This ensures that when the checkbox of Force enable zoom is checked, the web page will not be stretched.
This is the same that when the checkbox is unchecked.
Refer to: https://prometheanworld.atlassian.net/browse/AP9-2791

---
 content/browser/web_contents/web_contents_impl.cc | 2 +-
 components/browser_ui/accessibility/android/java/res/xml/accessibility_preferences.xml | 4 ++--
 components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java | 8 ++++----
 3 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -2907,9 +2907,9 @@
 #if BUILDFLAG(IS_ANDROID)
   for (FrameTreeNode* node : primary_frame_tree_.Nodes()) {
     RenderFrameHostImpl* rfh = node->current_frame_host();
     if (rfh->is_local_root()) {
       if (auto* rwh = rfh->GetRenderWidgetHost())
-        rwh->SetForceEnableZoom(web_preferences_->force_enable_zoom);
+        rwh->SetForceEnableZoom(false);
     }
   }
 #endif
diff --git a/components/browser_ui/accessibility/android/java/res/xml/accessibility_preferences.xml b/components/browser_ui/accessibility/android/java/res/xml/accessibility_preferences.xml
--- a/components/browser_ui/accessibility/android/java/res/xml/accessibility_preferences.xml
+++ b/components/browser_ui/accessibility/android/java/res/xml/accessibility_preferences.xml
@@ -13,14 +13,14 @@
     <org.chromium.components.browser_ui.accessibility.TextScalePreference
         android:key="text_scale"
         android:title="@string/font_size"
         android:selectable="false" />

-    <org.chromium.components.browser_ui.settings.ChromeBaseCheckBoxPreference
+    <!--<org.chromium.components.browser_ui.settings.ChromeBaseCheckBoxPreference
         android:key="force_enable_zoom"
         android:summary="@string/force_enable_zoom_summary"
-        android:title="@string/force_enable_zoom_title" />
+        android:title="@string/force_enable_zoom_title" />-->

     <org.chromium.components.browser_ui.settings.ChromeBaseCheckBoxPreference
         android:key="reader_for_accessibility"
         android:summary="@string/reader_for_accessibility_summary"
         android:title="@string/reader_for_accessibility_title" />
diff --git a/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java b/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java
--- a/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java
+++ b/components/browser_ui/accessibility/android/java/src/org/chromium/components/browser_ui/accessibility/AccessibilitySettings.java
@@ -33,6 +33,6 @@
     static final String PREF_FORCE_TABLET_UI = "force_tablet_ui";
     static final String PREF_MOVE_TOOLBAR_TO_BOTTOM = "move_toolbar_bottom";

     private TextScalePreference mTextScalePref;
-    private ChromeBaseCheckBoxPreference mForceEnableZoomPref;
+    //private ChromeBaseCheckBoxPreference mForceEnableZoomPref;
     private boolean mRecordFontSizeChangeOnStop;
@@ -45,11 +45,11 @@
     private FontSizePrefsObserver mFontSizePrefsObserver = new FontSizePrefsObserver() {
         @Override
         public void onFontScaleFactorChanged(float fontScaleFactor, float userFontScaleFactor) {
             mTextScalePref.updateFontScaleFactors(fontScaleFactor, userFontScaleFactor, true);
         }

         @Override
         public void onForceEnableZoomChanged(boolean enabled) {
-            mForceEnableZoomPref.setChecked(enabled);
+            //mForceEnableZoomPref.setChecked(enabled);
         }
     };
@@ -79,11 +79,11 @@
         mTextScalePref = (TextScalePreference) findPreference(PREF_TEXT_SCALE);
         mTextScalePref.setOnPreferenceChangeListener(this);
         mTextScalePref.updateFontScaleFactors(mFontSizePrefs.getFontScaleFactor(),
                 mFontSizePrefs.getUserFontScaleFactor(), false);

-        mForceEnableZoomPref =
+        /*mForceEnableZoomPref =
                 (ChromeBaseCheckBoxPreference) findPreference(PREF_FORCE_ENABLE_ZOOM);
         mForceEnableZoomPref.setOnPreferenceChangeListener(this);
-        mForceEnableZoomPref.setChecked(mFontSizePrefs.getForceEnableZoom());
+        mForceEnableZoomPref.setChecked(mFontSizePrefs.getForceEnableZoom());*/

         ChromeBaseCheckBoxPreference readerForAccessibilityPref =
--
2.17.1